march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -316,4 +305,25 @@ void MarchHardwareInterface::updatePowerNet()
     }
   }
 }
+
+bool MarchHardwareInterface::isOutSideLimits(int joint_index)
```

Comment:
The prefix `is` implies it returns a boolean. This is not used whenever you call it as a false gets never returned. Maybe rename it? Or throw the exception based on the return value?